### PR TITLE
[ED-308] Modify api_client to import all vulnerabilities when no asset IDs given and update documentation.

### PR DESCRIPTION
**Pull request by Dylan-OB**
**Description**

- The api_client.py file inside the api_edgescan directory has been modified to now include functionality which imports all findings when no asset IDs are given by the user.
- Edgescan's .md file has been updated. The previous information in the file has now been broken down into steps and explained in more depth. As well as this, I have added information regarding how to upload all assets (as discussed in the previous point) and how to avoid the duplication of findings.
- The parser.py file inside the api_edgescan directory has been modified to inform the user that leaving the "Service key 1" field blank will result in all asset's findings being imported.

**Test results**

- Each of the changes made were tested.

**Documentation**

- Documentation has been updated to include any additional information that was required.

### CODE REVIEW

**Review by ShayVD — APPROVED** [review]

File: dojo/tools/api_edgescan/parser.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def requires_tool_type(self, scan_type):
         return "Edgescan"
 
     def api_scan_configuration_hint(self):
-        return "the field <b>Service key 1</b> has to be set with the Edgescan asset id."
+        return "In the field <b>Service key 1</b>, provide the Edgescan asset ID(s). Leaving it blank will import all asset's findings."
```
Comment (cneill) [inline dojo/tools/api_edgescan/parser.py:32]:
```suggestion
        return "In the field <b>Service key 1</b>, provide the Edgescan asset ID(s). Leaving it blank will import all assets' findings."
```

**Review by cneill — CHANGES_REQUESTED** [review]
Just a tiny suggestion on the hint text, otherwise looks good to me

**Review by ShayVD — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Comment by cneill** [comment]
Could you please rebase this PR against `dev` again? There was an issue with the unit tests that has since been fixed

**Comment by ShayVD** [comment]
@cneill I've rebased the branch to the most up-to-date dev branch but 1 or 2 unit tests seem to be failing outright. Can you advise here, thanks.

**Review by cneill — CHANGES_REQUESTED** [review]
It looks like [this unit test](https://github.com/DefectDojo/django-DefectDojo/blob/a465e29bc50e4dd8aa71cb7085fe3b052d1d07a6/unittests/test_tool_config.py#L39) is broken by these changes. 

Updating to the following ought to fix it:
```
            self.assertEqual(acsh[i]['hint'], 'In the field <b>Service key 1</b>, provide the Edgescan asset ID(s). Leaving it blank will import all assets' findings.')
```

```
2023-11-10T09:57:04.0105570Z [36muwsgi_1         |[0m ======================================================================
2023-11-10T09:57:04.0106015Z [36muwsgi_1         |[0m FAIL: test_base (unittests.test_tool_config.TestApiScanConfigEntry.test_base) [Edgescan]
2023-11-10T09:57:04.0106299Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2023-11-10T09:57:04.0106514Z [36muwsgi_1         |[0m Traceback (most recent call last):
2023-11-10T09:57:04.0106853Z [36muwsgi_1         |[0m   File "/app/unittests/test_tool_config.py", line 39, in test_base
2023-11-10T09:57:04.0107384Z [36muwsgi_1         |[0m     self.assertEqual(acsh[i]['hint'], 'the field <b>Service key 1</b> has to be set with the Edgescan asset id.')
2023-11-10T09:57:04.0108100Z [36muwsgi_1         |[0m AssertionError: "In the field <b>Service key 1</b>, provi[75 chars]ngs." != 'the field <b>Service key 1</b> has to be[28 chars] id.'
2023-11-10T09:57:04.0108680Z [36muwsgi_1         |[0m - In the field <b>Service key 1</b>, provide the Edgescan asset ID(s). Leaving it blank will import all assets' findings.
2023-11-10T09:57:04.0109044Z [36muwsgi_1         |[0m + the field <b>Service key 1</b> has to be set with the Edgescan asset id.
2023-11-10T09:57:04.0109162Z [36muwsgi_1         |[0m 
2023-11-10T09:57:04.0109280Z [36muwsgi_1         |[0m 
2023-11-10T09:57:04.0109556Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
```

**Review by ShayVD — APPROVED** [review]

**Comment by ShayVD** [comment]
@cneill Should be all sorted now!

**Review by cneill — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Pull request merged**